design/one-pager-managed-resource-api-design.md

```diff
@@ -169,6 +173,79 @@ For both `Status` and `Spec`:
   What if the sub-resource is not yet supported as managed resource in Crossplane? In that case, you should first
   consider implementing that managed resource, if not suitable, only then include it in the CR.
 
+#### Embedded Structs with Mixed Fields
+
+Some provider APIs include an embedded struct that may contain some fields that
+are appropriate for `spec.forProvider` and some that are meant for
+`status.atProvider`. For example:
+
+```go
+// This is the provider's representation of the API object
+type ProviderAPIObject struct {
+  // Configurable field in top-level object
+  FieldOne *string `json:"fieldOne,omitempty"`
+
+  // Non-Configurable field in top-level object
+  FieldTwo *string `json:"fieldTwo,omitempty"`
+  
+  // Embedded struct in top-level object
+  EmbeddedStructOne *EmbeddedStruct `json:"embeddedStructOne,omitempty"`
+}
+
+type EmbeddedStruct struct {
+  // This field is configurable so it should be in spec.forProvider
+  SomeConfigurableField *string `json:"someConfigurableField,omitempty"`
+
+  // This field is configurable so it should be in spec.forProvider
+  AnotherConfigurableField *string `json:"anotherConfigurableField,omitempty"`
+
+  // This field is not configurable so it should be in status.atProvider
+  SomeNonConfigurableField string `json:"someNonConfigurableField,omitempty"`
+}
+```
+
+In this case, the solution is to divide the embedded struct into
+`EmbeddedStructSpec` and `EmbeddedStructStatus`.
```

Comment:
Would `<T>` and `<T>Observation` be more in line with our existing patterns? For example https://github.com/crossplaneio/stack-gcp/blob/d45813c/apis/database/v1beta1/cloudsql_instance_types.go#L92

Reply:
I could go either way. @muvaf suggested the `___Spec` and `___Status` pattern here, but I had also thought of the `Observed___` pattern in PR body [here](https://github.com/crossplaneio/stack-gcp/pull/111). I agree that `___Observation` more clearly matches the encompassing struct.

Reply:
I am actually realizing the `Spec` / `Status` pattern has already been implemented. It could be updated without modifying the API though.

https://github.com/crossplaneio/stack-gcp/blob/f7522e08bccbf856cb1cd3ad713efe597f45785b/apis/database/v1beta1/cloudsql_instance_types.go#L168

Thoughts?

Reply:
I'm happy with spec and status as long as it doesn't go against the status quo.